src/components/Quiz.js

```diff
@@ -18,19 +18,56 @@ const Quiz = () => {
     setQuestions(quizData);
   }, [quizName]);
 
-  const handleAnswerSelect = e => {
-    const selectedAnswer = { [e.target.name]: Number(e.target.value) };
+  const handleCheckboxAnswers = (e, answers) => {
+    const questionIds = Object.keys(answers || {});
+    const hasQPreviouslySelectedAnswers =
+      questionIds && questionIds.includes(e.target.name);
+
+    let checkedAnswer;
+    if (hasQPreviouslySelectedAnswers) {
```

Comment:
Since we don't need the `hasQPreviouslySelectedAnswers` variable, we can remove this `if` statement because line 41 will do the same job as line 45

Reply:
I was confused since I was not thinking out off the box bounding myself in the if statement. Thank you for helping me to see out of the box.